aioftp/client.py

```diff
@@ -785,17 +785,21 @@ async def __anext__(cls):
                 if cls.stream is None:
                     cls.stream = await cls._new_stream(path)
                 while True:
-                    line = await cls.stream.readline()
-                    while not line:
-                        await cls.stream.finish()
-                        if cls.directories:
-                            current_path, info = cls.directories.popleft()
-                            cls.stream = await cls._new_stream(current_path)
-                            line = await cls.stream.readline()
-                        else:
-                            raise StopAsyncIteration
-
-                    name, info = cls.parse_line(line)
+                    continue_reading = True
+                    while continue_reading:
+                        line = await cls.stream.readline()
+                        while not line:
+                            await cls.stream.finish()
+                            if cls.directories:
+                                current_path, info = cls.directories.popleft()
+                                cls.stream = await cls._new_stream(current_path)
+                                line = await cls.stream.readline()
+                            else:
+                                raise StopAsyncIteration
+
+                        name, info = cls.parse_line(line)
+                        # skipping . and .. as these are symlinks in Unix 
+                        continue_reading = str(name) in (".", "..")
```

Comment:
It is simpler not to do nested loop. You can just add check.
```suggestion
                    if str(name) in (".", ".."):
                        continue
```